pkg/network/flag/networkflag.go

```diff
@@ -0,0 +1,51 @@
+package flag
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+	"github.com/weaveworks/ignite/pkg/network"
+)
+
+var plugins = network.ListPlugins()
+
+type NetworkPluginFlag struct {
+	value *network.PluginName
+}
+
+func (nf *NetworkPluginFlag) Set(val string) error {
+	var foundPlugin *network.PluginName
+	for _, plugin := range plugins {
+		if plugin.String() == val {
+			foundPlugin = &plugin
+			break
+		}
+	}
+	if foundPlugin == nil {
+		return fmt.Errorf("Invalid network mode %q, must be one of %v", val, plugins)
+	}
+	*nf.value = *foundPlugin
+	return nil
+}
+
+func (nf *NetworkPluginFlag) String() string {
+	if nf.value == nil {
+		return ""
+	}
+	return nf.value.String()
+}
+
+func (nf *NetworkPluginFlag) Type() string {
+	return "network-mode"
```

Comment:
I would call this `plugin`, so the flag would be `--network-plugin plugin` instead of `--network-plugin network-mode`